### PR TITLE
Add support for signed cookies

### PR DESCRIPTION
**Pull request by darrendao**

### CODE REVIEW

**Comment by novaedge** [comment]
+1 Please merge! I support support for signed cookies.

**Comment by darrendao** [comment]
Anything else I can do to help get this merged?

**Comment by Laurian** [comment]
+1 Please merge, I need this :)

File: lib/cloudfrontUtil.js [hunk 1]
```diff
@@ -44,9 +82,9 @@ function getSignedUrl(cfUrl, params) {
   parsedUrl = url.parse(cfUrl, true);
   parsedUrl.search = '';
   _.extend(parsedUrl.query, {
-    'Expires': policy.expireTime,
-    'Policy': new Buffer(policy.toJSON()).toString('base64'),
-    'Signature': normalizeSignature(signature),
+    'Expires': params.expireTime / 1000,
+    'Policy': policy,
+    'Signature': sig,
```
Comment (jasonsims) [inline lib/cloudfrontUtil.js]: I like the addition of `genPolicy` but why the changes to `Expires` and `Signature`? It doesn't look like there needs to be this many modifications to `getSignedUrl`. Can you minimize the changes to this function? I think all that's needed is to update

`'Policy': new Buffer(policy.toJSON()).toString('base64')` ==> `'Policy': genPolicy(cfUrl, params)`

File: lib/cloudfrontUtil.js [hunk 2]
```diff
@@ -92,13 +151,13 @@ function getSignedRTMPUrl(domainname, s3key, params) {
...
-function normalizeSignature(sig) {
+function safeBase64(str) {
   var badCharMap = {'+': '-', '=': '_', '/': '~'};
   Object.keys(badCharMap).forEach(function (badChar) {
     var re = new RegExp('\\' + badChar, 'g');
-    sig = sig.replace(re, badCharMap[badChar]);
+    str = str.replace(re, badCharMap[badChar]);
   });
-  return sig;
+  return str;
```
Comment (jasonsims) [inline lib/cloudfrontUtil.js]: Why does this function have to be modified? There's no functional changes here.

**Comment by jasonsims** [comment]
Thanks for this @darrendao! Sorry for the delay. I just had a couple comments.

**Comment by reidmweber** [comment]
I've used this and have been able to get a few requests to work. Is there a reason you aren't setting the CloudFront-Expires cookie?

**Comment by saamo** [comment]
@jasonsims, how is going with this PR? It would be great if you merged it!

**Comment by jasonsims** [comment]
Hi everyone, I apologize for the delay on this one as it seems many people have been in need of this feature. @darrendao's changes have been incorporated within 1559a8f8d691cca560cd063cf7bdca5ff746d4cf which is now checked into master. I will publish a new release to NPM later today. I appreciate your patience and support for this repo!

**Comment by jasonsims** [comment]
This feature is now included in v2.1.0
